berkdb/lock/lock.c

```diff
@@ -4091,32 +4099,24 @@ __lock_getlocker_int(lt, locker, indx, partition, create, retries, retp,
 	if (sh_locker && create) {
 		sh_locker->tid = pthread_self();
 
+		sh_locker->snap_info = NULL;
 		if (gbl_print_deadlock_cycles) {
 			extern __thread snap_uid_t *osql_snap_info; /* contains cnonce */
-			if(osql_snap_info) sh_locker->snap_info = osql_snap_info;
-			else sh_locker->snap_info = NULL;
+			if(osql_snap_info)
+				sh_locker->snap_info = osql_snap_info;
```

Comment:
Not sure if sh_locker->snap_info is reset, since you get a cached one seems more robust to keep setting it to NULL explicitely

Reply:
Line 4102 sets it to null regardless what gbl_print_deadlock_cycles is, so this way is more reliable.